pkg/loop/internal/ccip/test/exec_provider.go

```diff
@@ -60,106 +65,169 @@ type staticExecProvider struct {
 }
 
 // ChainReader implements ExecProviderEvaluator.
-func (s *staticExecProvider) ChainReader() types.ChainReader {
+func (s staticExecProvider) ChainReader() types.ChainReader {
 	return nil
 }
 
 // Close implements ExecProviderEvaluator.
-func (s *staticExecProvider) Close() error {
+func (s staticExecProvider) Close() error {
 	return nil
 }
 
 // Codec implements ExecProviderEvaluator.
-func (s *staticExecProvider) Codec() types.Codec {
+func (s staticExecProvider) Codec() types.Codec {
 	return nil
 }
 
 // ContractConfigTracker implements ExecProviderEvaluator.
-func (s *staticExecProvider) ContractConfigTracker() libocr.ContractConfigTracker {
+func (s staticExecProvider) ContractConfigTracker() libocr.ContractConfigTracker {
 	return s.contractTracker
 }
 
 // ContractTransmitter implements ExecProviderEvaluator.
-func (s *staticExecProvider) ContractTransmitter() libocr.ContractTransmitter {
+func (s staticExecProvider) ContractTransmitter() libocr.ContractTransmitter {
 	return s.contractTransmitter
 }
 
 // Evaluate implements ExecProviderEvaluator.
-func (s *staticExecProvider) Evaluate(ctx context.Context, other types.CCIPExecProvider) error {
+func (s staticExecProvider) Evaluate(ctx context.Context, other types.CCIPExecProvider) error {
+	// OnRampReader test case
 	otherOnRamp, err := other.NewOnRampReader(ctx, "ignored")
 	if err != nil {
 		return fmt.Errorf("failed to create other on ramp reader: %w", err)
 	}
-	err = s.onrampreader.Evaluate(ctx, otherOnRamp)
+	err = s.onRampReader.Evaluate(ctx, otherOnRamp)
 	if err != nil {
-		return fmt.Errorf("on ramp reader evaluation failed: %w", err)
+		return evaluationError{err: err, component: onRampComponent}
 	}
+
+	// OffRampReader test case
+	// BCF-2991: offramp blocked on gas estimator
+	/*
+		otherOffRamp, err := other.NewOffRampReader(ctx, "ignored")
+		if err != nil {
+			return fmt.Errorf("failed to create other off ramp reader: %w", err)
+		}
+		err = s.offRampReader.Evaluate(ctx, otherOffRamp)
+		if err != nil {
+			return evaluationError{err: err, component: offRampComponent}
+		}
+	*/
```

Comment:
Restore?

Reply:
no, this will be a quick followup. it's not ready yet, i had an oversight earlier. i could delete it altogether, but expect to have it implemented in the next 1-2 days